lib/comgr/src/comgr-compiler.cpp

```diff
@@ -581,6 +582,23 @@ InProcessDriver::InProcessDriver(raw_ostream &DiagOS)
 }
 
 amd_comgr_status_t InProcessDriver::Execute(ArrayRef<const char *> Args) {
+  // For HIP compilation, we launch a process.
+  bool IsHIP = false;
+  for (auto &Arg : Args) {
+    if (StringRef(Arg) == "--genco") {
+      IsHIP = true;
+    }
+  }
+  if (IsHIP) {
+    std::stringstream SS;
+    SS << "hipcc ";
+    for (auto &Arg : Args) {
+      SS << " " << Arg;
```

Comment:
Does this handle arguments with embedded spaces?

Reply:
user is responsible to quote arguments containing spaces. we will pass the argument as it is.

Reply:
Currently Comgr is broken in this regard, as it just splits on spaces. I have a patch to fix this, but there is some more work to do as it is a backwards-incompatible change. Either way, we shouldn't require embedded quotes in the option strings, as it assumes a specific shell/command-line parser.

Rather than flatten the array of arguments to a single string and call `system` we should leave the array intact and use something like [llvm::sys::ExecuteAndWait](http://llvm.org/doxygen/namespacellvm_1_1sys.html#a98ac8572b014087563e9ff84ea048556). This way we don't hit cross-platform weirdness with `system`, we aren't open to any bugs or security issues with the shell parsing done by `system`, and every Comgr API treats the options the same.

Reply:
I agree with @scott-linder on the need to fix how arguments are passed. What state is your patch in so Sam's changes can be baased on the new approach?

Reply:
The Comgr change is ready to apply, I just need to coordinate the changes to all of the existing users of Comgr.

I thing Sam's change shouldn't need to worry about that, though. If we treat `Args` here as a shell/OS agnostic array of strings (each argument is a different element, and no characters need to be escaped or quoted) then it naturally supports both the existing (broken) implementation as well as the corrected version.